docs/topics/multiplatform-mobile/multiplatform-mobile-upgrade-app.md

```diff
@@ -0,0 +1,348 @@
+
+* [`kotlinx.coroutines`](https://github.com/Kotlin/kotlinx.coroutines) for using coroutines to write asynchronous code, thus allowing simultaneous operations
+* [`kotlinx.serialization`](https://github.com/Kotlin/kotlinx.serialization) for deserializing JSON responses into objects of entity classes used to process
+network operations
+* [Ktor](https://ktor.io/) framework as an HTTP client for retrieving data over the internet
+
+### kotlinx.coroutines
+
+To add `kotlinx.coroutines` to your project, specify a dependency in the common source set. For that, add the following
+line to the `build.gradle.kts` file of the shared module:
+
+```kotlin
+sourceSets {
+    commonMain {
+        dependencies {
+            // ...
+           implementation("org.jetbrains.kotlinx:kotlinx-coroutines-core:1.6.2")
+        }
+    }
+}
+```
+
+Multiplatform Gradle plugin automatically adds a dependency to the platform-specific (iOS and Android) parts
+of `kotlinx.coroutines`.
+
+You'll also use the new memory manager for Kotlin/Native, which will become default soon.
+Enable it in the same `build.gradle.kts` file:
+
+```kotlin
+val commonMain by getting {
+    // ...
+}
+
+kotlin.targets.withType(org.jetbrains.kotlin.gradle.plugin.mpp.KotlinNativeTarget::class.java) {
+    binaries.all {
+        binaryOptions["memoryModel"] = "experimental"
+    }
+}
+```
+
+### kotlinx.serialization
+
+For `kotlinx.serilization`, you need the plugin required by the build system. The Kotlin serialization plugin is shipped
+with the Kotlin compiler distribution, and the IntelliJ IDEA plugin is bundled into the Kotlin plugin.
+
+You can set up the serialization plugin with the Kotlin plugin using the Gradle plugins DSL by adding this line to
+the `plugins` block at the very beginning of the `build.gradle` file in the shared module:
+
+```kotlin
+plugins {
+    // 
+    kotlin("plugin.serialization") version "1.6.21"
+}
+```
+
+### Ktor
+
+You can add Ktor the same way you've added the `kotlinx.coroutines` library. In addition to specifying the core
+dependency in the common source set, you also need to:
+
+* Add the ContentNegotiation plugin with a special artifact to use `kotlinx.serialization` to deserialize JSON data into
+  a data class when receiving responses.
+* Provide the platform engines by adding dependencies on the corresponding artifacts in the platform source sets.
+
+```kotlin
+val ktorVersion = "2.0.2"
+
+sourceSets {
+    val commonMain by getting {
+        dependencies {
+            // ...
+            implementation("io.ktor:ktor-client-core:$ktorVersion")
+            implementation("io.ktor:ktor-client-content-negotiation:$ktorVersion")
+            implementation("io.ktor:ktor-serialization-kotlinx-json:$ktorVersion")
+        }
+    }
+    val androidMain by getting {
+        dependencies {
+            implementation("io.ktor:ktor-client-android:$ktorVersion")
+        }
+    }
+    val iosMain by creating {
```

Comment:
As a reader, I'd expect a small comment about why for ios it's 'by creating' and for everything else it's 'by getting'